src/revivalpmmp/pureentities/entity/WalkingEntity.php

```diff
@@ -222,49 +223,80 @@ protected function checkJump($dx, $dz){
-				PureEntities::logOutput("$this: checkJump(): cannot pass through the upper blocks!", PureEntities::DEBUG);
-				$this->yaw = $this->getYaw() + mt_rand(-120, 120) / 10;
+			
+		
+
+			$blockBox = $block->getCollisionBoxes()[0] ?? null;
+			$blockBoxDiff = $blockBox === null ? 0 : $blockBox->maxY - $blockBox->minY;
+			//Do the same for the upper block, id there is any collisionBox, consider as a block
+			//If i jump and have 2 slab or stair, so i cannot jump over these 2
+			PureEntities::logOutput("$this: CollisionBlock : $block --Height: $blockBoxDiff-- Pos: [x:" . $block->x . "] [y:" . $block->y . "] [z:" . $block->z . "]");
+			$upperblock = $this->getLevel()->getBlock($block->add(0, 1, 0));
+			if (!empty($upperblock->getCollisionBoxes())) {
+				$blockBoxDiff = $blockBoxDiff + 1;
+				PureEntities::logOutput("$this: CollisionBlock - upperBlock Detected - New Height($blockBoxDiff) : $upperblock - Pos: [x:" . $upperblock->x . "] [y:" . $upperblock->y . "] [z:" . $upperblock->z . "]");
 			}
-		}else{
-			PureEntities::logOutput("$this: checkJump(): no need to jump. Block can be passed! [canPassThrough:" . $blockingBlock->canPassThrough() . "] " .
-				"[jumpHeight:" . $this->getMaxJumpHeight() . "] [checkedBlock:" . $blockingBlock . "]", PureEntities::DEBUG);
+			if ($blockBoxDiff > $highestBlock) $highestBlock = $blockBoxDiff;
+			if ($lowestBlock == null) $lowestBlock = $blockBoxDiff;
+			elseif ($lowestBlock > $blockBoxDiff) $lowestBlock = $blockBoxDiff;
+			
+		}
+		//TODO: If one of the block i'm colliding is ok, could i jump on it ?
+		//Tested currently with 'lowestBlock', but if i want to jump over 2 block, and the block of the left or rigth (colliding)
+		//is OK for jump, monster will go up ... MAgic :D
+		PureEntities::logOutput("$this: CollisionBlock : highestBlock: $highestBlock");
+		if ($lowestBlock > 0 && $lowestBlock <=0.5 && $lowestBlock <= $this->getMaxJumpHeight()) {
+			// STAIR jump					
+			$this->motion->y = $this->gravity * 4;
+			PureEntities::logOutput("$this: highestBlock($highestBlock)/lowestBlock($lowestBlock): found slab or stair!", PureEntities::DEBUG);	
+			return true;
+		}
+		elseif ($lowestBlock > 0 && $lowestBlock <= $this->getMaxJumpHeight()) {
+			//Normal jump
+			PureEntities::logOutput("$this: highestBlock($highestBlock)lowestBlock($lowestBlock): Normal Jump - set motion to gravity * 3.2!", PureEntities::DEBUG);
+			$this->motion->y = $this->gravity * 3.2;
+			return true;
+		}
+		elseif ($lowestBlock > 0 && $lowestBlock > $this->getMaxJumpHeight()) {
+			PureEntities::logOutput("$this: highestBlock($highestBlock)lowestBlock($lowestBlock): cannot pass through the upper blocks!", PureEntities::DEBUG);
+			$this->yaw = $this->getYaw() + mt_rand(-120, 120) / 10;
+		}
+		else {
+			PureEntities::logOutput("$this: highestBlock($highestBlock)lowestBlock($lowestBlock): no need to jump. Block can be passed!");
 		}
 		return false;
+		
 	}
+	
+    public function getLookingBlock() {
```

Comment:
Nothing from this function is used.  It should be removed.

Reply:
Done